estimation/observations.R

```diff
@@ -3,213 +3,88 @@ library(strafica)
-
-background_information = rbind_list(background_information, df)
-
-# Add personal data to tours
-
-stopifnot(length(unique(background_information$pid)) == nrow(background_information))
-observations = leftjoin(observations, background_information)
-
-# Divide data into two models
-
-m = match(observations$rzone, zones$zone)
-hs15_model = (zones$capital_region[m] | zones$surrounding_municipality[m]) &
-    observations$izone %in% zones$zone &
-    observations$jzone %in% zones$zone &
-    observations$survey == 0
-outer_model = zones$peripheral_municipality[m] &
-    observations$izone %in% zones$zone &
-    observations$jzone %in% zones$zone &
-    observations$survey == 2
-stopif(any(hs15_model & outer_model))
-
-observations$model = NA
-observations$model[hs15_model] = 1
-observations$model[outer_model] = 2
-
-# Subset estimation data
-
-observations = subset(observations, mode %in% 1:5)
-observations = subset(observations, !is.na(model))
-
-observations = downclass(observations)
-check.na(observations)
-save(observations, file="observations.RData")
+for (i in rows.along(input)) {
+    
+    message(sprintf("%d/%d: %s", i, nrow(input), input$file[i]))
+    tours = load1(input$file[i])
+    
+    observations = data.frame(pid=tours$pid)
+    observations$mode = tours$mode
+    observations$ttype = NA
+    m = grepl("^(1 - 2)", tours$tour_type, perl=TRUE)
+    observations$ttype[m] = 1
+    m = grepl("^(1 - 3)", tours$tour_type, perl=TRUE)
+    observations$ttype[m] = 2
+    m = grepl("^(1 - 4)", tours$tour_type, perl=TRUE)
+    observations$ttype[m] = 3
+    m = grepl("^(1 - 5)", tours$tour_type, perl=TRUE)
+    observations$ttype[m] = 4
+    m = grepl("^1$|^(1 - [16])", tours$tour_type, perl=TRUE)
+    observations$ttype[m] = 5
+    m = grepl("^2$|^(2 - [123456])", tours$tour_type, perl=TRUE)
+    observations$ttype[m] = 6
+    m = is.na(observations$ttype)
+    observations$ttype[m] = 7
+    m = (observations$constructed & observations$ttype == 1)
+    observations$ttype[m] = 8
+    m = (observations$constructed & observations$ttype != 8)
+    observations$ttype[m] = 9
+    
+    observations$other_destinations = ifelse(rowSums(tours[, grepl("^visits_t", colnames(tours), perl=TRUE)]) > 2, 1, 0)
+    observations$closed = ifelse(tours$closed, 1, 2)
+    
+    m = match(tours$zone_origin, zones$zone_orig)
+    observations$izone = zones$zone[m]
+    observations$izone_cbd = ifelse(zones$cbd[m]==1, 1, 0)
+    observations$izone_population_density = zones$population_density[m]
+    observations$izone_housing = zones$housing[m]
+    observations$izone_parking_fee_other = zones$parking_fee_other[m]
+    observations$izone_cars_per_people = zones$cars_per_people[m]
+    
+    m = match(tours$zone_destination, zones$zone_orig)
+    observations$jzone = zones$zone[m]
+    
+    observations$ipeak = apply.breaks(tours$itime_origin,
+                            class=c("morning","afternoon"),
+                            c("06:00:00","15:00:00"),
```

Comment:
Again a function for these would be a good idea, and perhaps define the time-constants somewhere. Put function could be enough.

Reply:
https://github.com/HSLdevcom/helmet-data-preprocessing/pull/12/commits/c3968799d020705d3cff7cee065dbe5afb9a86e9 Well, now the timestamps are defined inside the function. Should they still be constants somewhere? Should there be a way to manage constants, like reading them from a file?